internal/trace/context.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trace
+
+import (
+	"context"
+	"io/ioutil"
+
+	"github.com/sirupsen/logrus"
+)
+
+// loggerKey is the associated key type for logger entry in context.
+type loggerKey struct{}
```

Comment:
A custom-typed int logger key would be better as we can have a constant key so that we don't need to create an object each time.

Reply:
I am following the suggestion in [context doc](https://pkg.go.dev/context#:~:text=should%20not%20be%20of%20type%20string%20or%20any%20other%20built%2Din%20type%20to%20avoid%20collisions%20between%20packages%20using%20context) for the key